Automattic/wp-calypso
# Reader Lists: switch follow and unfollow to use the data layer

#### Changes proposed in this Pull Request

Modernize the code used to follow and unfollow a Reader list, using data layer instead of thunks and `wpcom.undocumented()`. I've tried to reduce use of lodash and also removed 'dismiss notice' functionality which is no longer in use. Yay for throwing away code.

#### Testing instructions



Find a public list like http://calypso.localhost:3000/read/list/blowery/comics and make sure you can follow/unfollow it successfully.

You can also run the unit tests with `yarn test-client client/state/reader/lists`.
